### PR TITLE
Use `missingType` instead of `undefinedType` for optional methods under `exactOptionalPropertyTypes`

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/52494

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot user test tsserver
@typescript-bot test tsserver top100
@typescript-bot run dt
@typescript-bot perf test this

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11235,7 +11235,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return baseTypeVariable ? getIntersectionType([type, baseTypeVariable]) : type;
         }
         else {
-            return strictNullChecks && symbol.flags & SymbolFlags.Optional ? getOptionalType(type) : type;
+            return strictNullChecks && symbol.flags & SymbolFlags.Optional ? getOptionalType(type, /*isProperty*/ true) : type;
```
Comment (jakebailey) [inline src/compiler/checker.ts:11238]: I'm honestly really suspicious of every other call to this function now; I went and made `isProperty` required and about _half_ of the callers are actually working with things that are explicitly property types.

No tests change when I make everyone pass in this parameter, though, which makes me think that we don't have enough coverage of this particular option.

Comment (Andarist) [reply src/compiler/checker.ts:11238]: This option has a default though - did you try passing `true` to all calls of this function?

Comment (jakebailey) [reply src/compiler/checker.ts:11238]: No, I mean I changed it from `isProperty = false` to `isProperty: boolean` and then examined all of the places where it error'd.

Not that this is directly required for this PR, just something I noticed.

Comment (Andarist) [reply src/compiler/checker.ts:11238]: The `missingType` thing is only relevant here for with `exactOptionalPropertyTypes` - and that's not the most popular compiler option. So I definitely wouldn't be surprised if the test suite wouldn't be that comprehensive in this regard.

**Review by weswigham — APPROVED** [review]
As jake says, we might need to audit our other usages of `getOptionalType` to be more intentional about passing the second parameter; but this does look like the correct fix for this case.

**Review by jakebailey — APPROVED** [review]

**Comment by jakebailey** [comment]
I'm going to merge this because it's correct, but, I do think we need to audit the rest and come up with some test cases.

**Pull request merged**